0003/README.md

```diff
@@ -0,0 +1,116 @@
+
+
+## Progress:
+
+This implementation of rsyslog has been verified with POC and been discussed in-depth within the Packet and Infracloud team.  
+
+## Suggestions:
+
+- We want to ensure at a basic minimum there is a centralized logging framework which exists for tinkerbell. In order to have a generic implementation of configuring log-drivers below are some proposals.
+
+1) How to make action containers run with logging drivers.
+
+a) Place the logging details inside the workflow template. 
+```
+	version: "0.1"
+	name: hello_world_workflow
+	global_timeout: 600
+	tasks:
+	  - name: "hello world"
+	    worker: "{{.device_1}}"
+	    actions:
+	      - name: "hello_world"
+		image: hello-world
+		timeout: 60  
+	logging:
+	  type: syslog  # log driver
+	  config:
+	    syslog-address : tcp://192.168.1.1:514
+	    tag: "container_name/{{.Name}}"
+ ```
+
+ - This will ensure all the worflow action containers have consistent way of logging informations. 
+ - logging configuration would be specific to workflow containers. 
+
+
+  b) Host level configuration. 
+- We can have a file for host level configuration. This will set environment variables which can be used in action.go while creating the containes. 
+- This will have a way where all the workflow template will have the common logging framework.
+- Problem
+    - Design issue: logging here is specific to action containers. So having them in the workflow template make more sense other than in separate config file. 
+    - maintaining multi-node cluster 
+    - updating the logging configuration. 
+
+
+
+ Workaround 
+   - Have configuration defined in action.go when the action container is created. This will ensure we have bare centralized framework ready.
+   - Challenge:  
+     - change in implementation when changes are done in provisioner. Like, moving the whole infrastructure on a private network. 
+
+## System Context Diagram
+![rsyslog](rsyslog.png)
+
+## Future roadmap:
+
+As a part of roadmap implementation, certain changes in the existing framework would be required to bring below proposal onboard.
+
+1) Provisioner/Worker configuration
+   Pushing logging configuration on the nodes via some template or config file. 
+
+2) Centralized logging in Private infrastructure of Provisioner
+
+3) Fault tolerance in logging. 
+- Handling of scenarios where communication can break over network or nodes can go down. 
+- Handling of scenarios where there can be distributed provisioner having services running across multiple nodes. 
+
+
+4) Log files structure:
+- Worker container Files to have workflow Id. In the current framework, this can be done. Changes would be required to attach id of workflow in the name of worker action containers. 
+- Files for each node will be placed in the respective directory on logging server.
+
+
+5) Implementation with fluentbit and other logging drivers:  
+* Separate doc on how rsyslog, fluentbit would work. 
+
+
+## Documentation:
+
```

Comment:
transitioning the content here to documentation should go pretty smoothly, this is an excellent write up.

Reply:
We need to cover those topics as part of the first release.

1 - We need to document a high level of how logging works, where it takes action, and what's the purpose and when how to use it.
2 - We should add as part of the Setup guides a reference to (1) + the concrete centralization logging strategy the setup uses. Vagrant will use rsyslog, so we will have to link rsyslog documentation to Vagrant setup.

Reply:
I agree, as a part of this change document will cover

Logging architecture to help the user understand our implementation.
How rsyslog is used
How other log-driver can be used.
In first phase of implementation, we will add point 1 and 2. After Tinkerbell as a product is ready where a user can change log-driver, we will add point 3, may be also with an example.